torchao/prototype/mx_formats/custom_cast.py

```diff
@@ -21,6 +21,8 @@
 
 from torchao.prototype.mx_formats.constants import (
     DTYPE_FP4,
+    DTYPE_FP4_E2M1,
```

Comment:
for all the changes in this file, can you rebase past https://github.com/pytorch/ao/pull/363 ?  

and in the code after that PR, perhaps we can add stochastic rounding option to `_f32_to_fpx_unpacked`?

for rounding mode, how about something like this instead of a boolean?

```
class RoundingMode(enum.Enum):
    TIE_TO_EVEN = auto()  # default
    STOCHASTIC = auto()  # added in this PR

def foo(..., rounding_mode=RoundingMode.TIE_TO_EVEN, ...): ...
```